evalml/automl/engine/engine_base.py

```diff
@@ -315,26 +348,107 @@ def train_and_score_pipeline(
         ):
             evaluation_entry["binary_classification_threshold"] = cv_pipeline.threshold
         cv_data.append(evaluation_entry)
-    training_time = time.time() - start
     cv_scores = pd.Series([fold["mean_cv_score"] for fold in cv_data])
     cv_score_mean = cv_scores.mean()
     logger.info(
         f"\tFinished cross validation - mean {automl_config.objective.name}: {cv_score_mean:.3f}",
     )
+
+    holdout_score = np.NaN
+    holdout_scores = np.NaN
+    if use_holdout:
+        logger.info("\tStarting holdout set scoring")
+        logger.debug(f"\t\tTraining and scoring entire dataset")
+        try:
+            logger.debug(f"\t\t\tFull training data pipeline: starting training")
+            full_pipeline, hashes = train_pipeline(
+                pipeline,
+                full_X_train,
+                full_y_train,
+                automl_config,
+                schema=False,
+                get_hashes=True,
+            )
+            stored_pipeline = full_pipeline
+            logger.debug(f"\t\t\tFull training data pipeline: finished training")
+            if (
+                automl_config.optimize_thresholds
+                and is_binary(automl_config.problem_type)
+                and full_pipeline.threshold is not None
+            ):
+                logger.debug(
+                    f"\t\t\tFull data pipeline: Optimal threshold found ({full_pipeline.threshold:.3f})",
+                )
+            logger.debug(f"\t\t\tScoring trained full training data pipeline")
+            holdout_scores = full_pipeline.score(
```

Comment:
Curious what is `objectives_to_score` here

Reply:
Might be more clear with the new private function reorg, but it should be the main objective + any other additional objectives specified by `automl.additional_objectives`.